specs/phase0/fork-choice.md

```diff
@@ -275,6 +289,55 @@ def get_head(store: Store) -> Root:
         head = max(children, key=lambda root: (get_latest_attesting_balance(store, root), root))
 ```
 
+#### `get_proposer_head`
+
+```python
+def get_proposer_head(store: Store, head_root: Root, slot: Slot) -> Root:
+    justified_state = store.checkpoint_states[store.justified_checkpoint]
+    head_block = store.blocks[head_root]
+    parent_root = head_block.parent_root
+    parent_block = store.blocks[parent_root]
+
+    # Only re-org a single slot at most.
+    single_slot_reorg = parent_block.slot + 1 == head_block.slot and head_block.slot + 1 == slot
+
+    # Do not re-org on an epoch boundary where the proposer shuffling could change.
+    shuffling_stable = slot % SLOTS_PER_EPOCH != 0
+
+    # Ensure that the FFG information of the new head will be competitive with the current head.
+    # FIXME: needs more justification/finalization information in the store
+    # ffg_competitive =
+    #    store.realized_justifications[parent_root] ==
+    #        store.realized_justifications[head_root] and
+    #    store.realized_finalizations[parent_root] ==
+    #        store.realized_finalization[head_root]
+    ffg_competitive = True
+
+    # Ensure that total participation meets a minimum threshold. The factor of 2 accounts for votes
+    # from the head slot which should flow to the parent either via the head block or votes for
+    # emptiness.
+    participation_multiplier = 2
+    parent_weight = get_latest_attesting_balance(store, parent_root)
+    participation_threshold = calculate_committee_fraction(justified_state, PARTICIPATION_THRESHOLD)
+    participation_ok = parent_weight >= participation_multiplier * participation_threshold
+
+    # Only re-org the head block if it arrived later than the attestation deadline.
+    head_late = store.block_timeliness.get(head_root) is False
+
+    # Check that the head has few enough votes to be overpowered by our proposer boost.
+    assert store.proposer_boost_root == Root()
```

Comment:
Doesn't this check serve the same purpose as `is_timely`? We could just use only this check instead of storing `is_timely` in the fork choice store.

Reply:
From reading your implementation, I realized this isn't the same because this will be called in the proposal slot, so proposer boost should be reset by this point

Reply:
yeah I intended this assert mainly as a sanity check, perhaps we could add a check on `get_current_slot(store) == slot` as well

Reply:
Verygood👌🙏🌹